replica/table.cc

```diff
@@ -2002,7 +1997,7 @@ void table::update_effective_replication_map(locator::effective_replication_map_
         if (new_tablet_count > old_tablet_count) {
             tlogger.info0("Detected tablet split for table {}.{}, increasing from {} to {} tablets",
                           _schema->ks_name(), _schema->cf_name(), old_tablet_count, new_tablet_count);
-            handle_tablet_split_completion(old_tablet_count, _erm->get_token_metadata().tablets().get_tablet_map(table_id));
+            co_await handle_tablet_split_completion(old_tablet_count, _erm->get_token_metadata().tablets().get_tablet_map(table_id));
         }
```

Comment:
Ah found a problem with coroutinizing

1) tablet map is updated

2) handling of split completion yields, before new storage groups are set

3) token is mapped to tablet id which storage group doesn't exist yet

what we can do is to defer tablet map update to after handling of split completion

we can rely on table::update_effective_replication_map keeping old erm alive till completion.

Reply:
okay, can do that